contracts/std-lib-extended/src/option.sw

```diff
@@ -0,0 +1,31 @@
+library option;
+
+impl<T> Option<T> {
+    pub fn expect<M>(self, message: M) -> T {
+        match self {
+            Option::Some(inner) => inner,
+            Option::None => {
+                // Revert with the message.
+                // The `revert` function only accepts u64, so as
+                // a workaround we use require.
+                require(false, message);
+                // Trick the compiler into knowing that the branch
+                // will revert and doesn't need to have the same return
+                // type as Option::Some
```

Comment:
not sure I follow this
why doesnt https://github.com/FuelLabs/sway/blob/6f5cca5cbea7d2d11fc12787db51c1a2071504e7/sway-lib-std/src/option.sw#L1 satisfy our needs?

Reply:
We could use `unwrap` here https://github.com/FuelLabs/sway/blob/6f5cca5cbea7d2d11fc12787db51c1a2071504e7/sway-lib-std/src/option.sw#L159 but the issue is it doesn't let you revert with a revert message, it just reverts with a super unhelpful code of `0`. Same with Result's `unwrap` https://github.com/FuelLabs/sway/blob/6f5cca5cbea7d2d11fc12787db51c1a2071504e7/sway-lib-std/src/result.sw#L149

So the options are:
1. Use the existing `unwrap` functions and it'll be much harder to debug when something reverts
2. Do something like:
```
require(my_option.is_some(), "my revert msg");
let my_value = my_option.unwrap();
```
The downside of this is now there are 2 match statements which is just wasteful (1 match statement in [is_some](https://github.com/FuelLabs/sway/blob/6f5cca5cbea7d2d11fc12787db51c1a2071504e7/sway-lib-std/src/option.sw#L107), and another in [unwrap](https://github.com/FuelLabs/sway/blob/6f5cca5cbea7d2d11fc12787db51c1a2071504e7/sway-lib-std/src/option.sw#L157). It also sucks to not have a 1 liner
3. Create an `expect` fn which imo is way more ergonomic to have a 1 liner that gives helpful revert messages